keras_retinanet/models/retinanet.py

```diff
@@ -148,31 +195,76 @@ def num_anchors(self):
 
 
 def default_submodels(num_classes, anchor_parameters):
+    """ Create a list of default submodels used for object detection.
+
+    The default submodels contains a regression submodel and a classification submodel.
+
+    Args
+        num_classes       : Number of classes to use.
+        anchor_parameters : Struct that defines how the anchors should be made.
+
+    Returns
+        A list of tuple, where the first element is the name of the submodel and the second element is the submodel itself.
+    """
     return [
         ('regression', default_regression_model(anchor_parameters.num_anchors())),
         ('classification', default_classification_model(num_classes, anchor_parameters.num_anchors()))
     ]
 
 
 def __build_model_pyramid(name, model, features):
+    """ Applies a single submodel to each FPN level.
+
+    Args
+        name     : Name of the submodel.
+        model    : The submodel to evaluate.
+        features : The FPN features.
+
+    Returns
+        A tensor containing the response from the submodel on the FPN features.
+    """
     return keras.layers.Concatenate(axis=1, name=name)([model(f) for f in features])
 
 
 def __build_pyramid(models, features):
+    """ Applies all submodels to each FPN level.
+
+    Args
+        models   : List of sumodels to run on each pyramid level (by default only regression, classifcation).
+        features : The FPN features.
+
+    Returns
+        A list of tensors, one for each submodel.
+    """
     return [__build_model_pyramid(n, m, features) for n, m in models]
 
 
 def __build_anchors(anchor_parameters, features):
-    anchors = []
-    for i, f in enumerate(features):
-        anchors.append(layers.Anchors(
+    """ Builds anchors for the shape of the features from FPN.
+
+    Args
+        anchor_parameters : Parameteres that determine how anchors are generated.
+        features          : The FPN features.
+
+    Returns
+        A tensor containing the anchors for the FPN features.
+
+        The shape is:
+        ```
+        (batch_size, num_anchors, 4)
+        ```
+    """
+    anchors = [
+        layers.Anchors(
             size=anchor_parameters.sizes[i],
             stride=anchor_parameters.strides[i],
             ratios=anchor_parameters.ratios,
             scales=anchor_parameters.scales,
             name='anchors_{}'.format(i)
-        )(f))
-    return keras.layers.Concatenate(axis=1)(anchors)
+        )(f) for i, f in enumerate(features)
+    ]
```

Comment:
While I like doing things in single expressions, I'm not so sure the list comprehension is really an improvement over the loop in terms of readability.

Reply:
I personally like list comprehensions, but it may reduce readability for some users. Although I do prefer not having an empty definition for anchors, it's all neatly packaged via list comp.